lib/cnet/punt/punt_ether_kernel.c

```diff
@@ -0,0 +1,185 @@
+punt_ether_kernel_process_mbuf(struct cne_node *node, pktmbuf_t **mbufs, uint16_t cnt)
+{
+    punt_ether_kernel_node_ctx_t *ctx = (punt_ether_kernel_node_ctx_t *)node->ctx;
+
+    if (ctx->sock >= 0) {
+        for (int i = 0; i < cnt; i++)
+            pktmbuf_adj_offset(mbufs[i], -(mbufs[i]->l2_len));
+
+        int nb = pktdev_tx_burst(ctx->lport, mbufs, cnt);
+        if (nb == PKTDEV_ADMIN_STATE_DOWN)
+            CNE_WARN("Failed to send packets: %s\n", strerror(errno));
+    }
+}
+
+static uint16_t
+punt_ether_kernel_node_process(struct cne_graph *graph __cne_unused, struct cne_node *node,
+                               void **objs, uint16_t nb_objs)
+{
+    uint16_t n_left_from;
+    pktmbuf_t *mbufs[PREFETCH_CNT], **pkts;
+    int k;
+
+    pkts        = (pktmbuf_t **)objs;
+    n_left_from = nb_objs;
+
+    for (k = 0; k < PREFETCH_CNT && k < n_left_from; k++)
+        cne_prefetch0(pktmbuf_mtod_offset(pkts[k], void *, sizeof(struct cne_ether_hdr)));
+
+    while (n_left_from >= PREFETCH_CNT) {
+        /* Prefetch next-next mbufs */
+        if (likely(n_left_from > ((PREFETCH_CNT * 3) - 1))) {
+            cne_prefetch0(pkts[(PREFETCH_CNT * 2) + 0]);
+            cne_prefetch0(pkts[(PREFETCH_CNT * 2) + 1]);
+            cne_prefetch0(pkts[(PREFETCH_CNT * 2) + 2]);
+            cne_prefetch0(pkts[(PREFETCH_CNT * 2) + 3]);
+            cne_prefetch0(pkts[(PREFETCH_CNT * 2) + 4]);
+            cne_prefetch0(pkts[(PREFETCH_CNT * 2) + 5]);
+        }
+
+        /* Prefetch next mbuf data */
+        if (likely(n_left_from > ((PREFETCH_CNT * 2) - 1))) {
+            uint16_t pre = PREFETCH_CNT;
+
+            cne_prefetch0(pktmbuf_mtod_offset(pkts[pre + 0], void *, pkts[pre + 0]->l2_len));
+            cne_prefetch0(pktmbuf_mtod_offset(pkts[pre + 1], void *, pkts[pre + 1]->l2_len));
+            cne_prefetch0(pktmbuf_mtod_offset(pkts[pre + 2], void *, pkts[pre + 2]->l2_len));
+            cne_prefetch0(pktmbuf_mtod_offset(pkts[pre + 3], void *, pkts[pre + 3]->l2_len));
+            cne_prefetch0(pktmbuf_mtod_offset(pkts[pre + 4], void *, pkts[pre + 4]->l2_len));
+            cne_prefetch0(pktmbuf_mtod_offset(pkts[pre + 5], void *, pkts[pre + 5]->l2_len));
+        }
+
+        memcpy(mbufs, pkts, (PREFETCH_CNT * sizeof(void *)));
+
+        pkts += PREFETCH_CNT;
+        n_left_from -= PREFETCH_CNT;
+
+        punt_ether_kernel_process_mbuf(node, mbufs, PREFETCH_CNT);
+    }
+
+    while (n_left_from > 0) {
+        mbufs[0] = pkts[0];
+
+        n_left_from--;
+        pkts++;
+
+        punt_ether_kernel_process_mbuf(node, mbufs, 1);
+    }
+
+    return nb_objs;
+}
+
+static int
+punt_ether_kernel_node_init(const struct cne_graph *graph __cne_unused, struct cne_node *node)
+{
+    punt_ether_kernel_node_ctx_t *ctx = (punt_ether_kernel_node_ctx_t *)node->ctx;
+
+    lport_cfg_t cfg = {0}; /**< CFG for tun/tap setup */
+    ctx->mmap       = mmap_alloc(DEFAULT_MBUF_COUNT, DEFAULT_MBUF_SIZE, MMAP_HUGEPAGE_4KB);
```

Comment:
can probably make this smaller... as we don't expect so many packets to flow through this interface